src/Compilers/CSharp/Test/IOperation/IOperation/IOperationTests_IFromEndIndexOperation_IRangeOperation.cs

```diff
@@ -0,0 +1,224 @@
+
+.locals {R1}
+{
+    Locals: [System.Range a] [System.Range b] [System.Range c] [System.Range d]
+    Block[B1] - Block
+        Predecessors: [B0]
+        Statements (4)
+            ISimpleAssignmentOperation (OperationKind.SimpleAssignment, Type: System.Range, IsImplicit) (Syntax: 'a = start..end')
+              Left: 
+                ILocalReferenceOperation: a (IsDeclaration: True) (OperationKind.LocalReference, Type: System.Range, IsImplicit) (Syntax: 'a = start..end')
+              Right: 
+                IRangeOperation (OperationKind.Range, Type: System.Range) (Syntax: 'start..end')
+                  LeftOperand: 
+                    IParameterReferenceOperation: start (OperationKind.ParameterReference, Type: System.Index) (Syntax: 'start')
+                  RightOperand: 
+                    IParameterReferenceOperation: end (OperationKind.ParameterReference, Type: System.Index) (Syntax: 'end')
+
+            ISimpleAssignmentOperation (OperationKind.SimpleAssignment, Type: System.Range, IsImplicit) (Syntax: 'b = start..')
+              Left: 
+                ILocalReferenceOperation: b (IsDeclaration: True) (OperationKind.LocalReference, Type: System.Range, IsImplicit) (Syntax: 'b = start..')
+              Right: 
+                IRangeOperation (OperationKind.Range, Type: System.Range) (Syntax: 'start..')
+                  LeftOperand: 
+                    IParameterReferenceOperation: start (OperationKind.ParameterReference, Type: System.Index) (Syntax: 'start')
+                  RightOperand: 
+                    null
+
+            ISimpleAssignmentOperation (OperationKind.SimpleAssignment, Type: System.Range, IsImplicit) (Syntax: 'c = ..end')
+              Left: 
+                ILocalReferenceOperation: c (IsDeclaration: True) (OperationKind.LocalReference, Type: System.Range, IsImplicit) (Syntax: 'c = ..end')
+              Right: 
+                IRangeOperation (OperationKind.Range, Type: System.Range) (Syntax: '..end')
+                  LeftOperand: 
+                    null
+                  RightOperand: 
+                    IParameterReferenceOperation: end (OperationKind.ParameterReference, Type: System.Index) (Syntax: 'end')
+
+            ISimpleAssignmentOperation (OperationKind.SimpleAssignment, Type: System.Range, IsImplicit) (Syntax: 'd = ..')
+              Left: 
+                ILocalReferenceOperation: d (IsDeclaration: True) (OperationKind.LocalReference, Type: System.Range, IsImplicit) (Syntax: 'd = ..')
+              Right: 
+                IRangeOperation (OperationKind.Range, Type: System.Range) (Syntax: '..')
+                  LeftOperand: 
+                    null
+                  RightOperand: 
+                    null
+
+        Next (Regular) Block[B2]
+            Leaving: {R1}
+}
+
+Block[B2] - Exit
+    Predecessors: [B1]
+    Statements (0)
+";
+
+            VerifyFlowGraphForTest<BlockSyntax>(compilation, expectedFlowGraph);
+        }
+    }
```

Comment:
>} [](start = 4, length = 1)

It feels like we need to add "standard" set of tests for control flow within IFromEndIndexOperation and IRangeOperation nodes. And tests for regions created for captures.

Reply:
Added a note to the follow up issue: [https://github.com/dotnet/roslyn/issues/31545#issuecomment-444579051](https://github.com/dotnet/roslyn/issues/31545#issuecomment-444579051).

---
In reply to: [238945524](https://github.com/dotnet/roslyn/pull/31546#discussion_r238945524) [](ancestors = 238945524)